src/azure-cli-testsdk/azure/cli/testsdk/checkers.py

```diff
@@ -23,7 +24,12 @@ def __call__(self, execution_result):
         except jmespath.exceptions.JMESPathTypeError:
             raise JMESPathCheckAssertionError(self._query, self._expected_result, actual_result,
                                               execution_result.output)
-        if actual_result != self._expected_result and str(actual_result) != str(self._expected_result):
+        if self._case_sensitive:
+            equals = actual_result == self._expected_result or str(actual_result) == str(self._expected_result)
+        else:
+            equals = isinstance(actual_result, str) and isinstance(self._expected_result, str)\
```

Comment:
what's the difference between `str(actual_result).lower() == str(self._expected_result).lower()` and your logic? if `self._case_sensitive==false`, `actual_result` and `self._expected_result` must be the type of string. Do I misunderstand?

Reply:
Good question. I have updated it.